libs/ui/src/screens/activity/activity.tsx

```diff
@@ -15,18 +22,31 @@ import {
 import { styles } from './activity.styles';
 
 export const Activity: FC = () => {
+  const { params } = useRoute<RouteProp<ScreensParamList, ScreensEnum.Activity>>();
+
   const { navigate } = useNavigation();
   const publicKeyHash = useSelectedAccountPublicKeyHashSelector();
   const { chainId } = useSelectedNetworkSelector();
 
   const navigateToWallet = () => navigate(ScreensEnum.Wallet);
+  const openActivityFilters = () =>
+    navigate(ScreensEnum.ActivityFilterSelector, { filterType: params?.filterType ?? ACTIVITIES_TYPES[0] });
+
+  const iconName =
+    isDefined(params) && params?.filterType.id !== ACTIVITIES_TYPES[0].id
+      ? IconNameEnum.FilterYes
+      : IconNameEnum.FilterNo;
 
   return (
     <ScreenContainer style={styles.root}>
       <HeaderContainer isSelectors>
         <ScreenTitle title="Transactions" onBackButtonPress={navigateToWallet} />
       </HeaderContainer>
-      <ActivitySectionList publicKeyHash={publicKeyHash} chainId={chainId} />
+      <Row style={styles.panel}>
+        <Text style={styles.filterName}>{params?.filterType.title ?? ACTIVITIES_TYPES[0].title}</Text>
+        <TouchableIcon name={iconName} onPress={openActivityFilters} />
+      </Row>
+      <ActivitySectionList publicKeyHash={publicKeyHash} chainId={chainId} filterType={params?.filterType.value} />
```

Comment:
you are using `??` twice and `?.` 4 times for `filterType`.  
you could just create a variable with fallback value and use it
```
const { params } = useRoute<RouteProp<ScreensParamList, ScreensEnum.Activity>>();
const filterType = params?.filterType ?? ACTIVITIES_TYPES[0];
```

this will remove a lot of code